test-data/unit/fine-grained-suggest.test

```diff
@@ -456,3 +456,31 @@ def bar() -> None:
 [out]
 (str) -> str
 ==
+
+[case testSuggestInferClassMethod]
+# flags: --strict-optional
+# suggest: foo.F.bar
+# suggest: foo.F.baz
+[file foo.py]
+class F:
+    @classmethod
+    def bar(cls, x, y):
+        return x
+
+    @staticmethod
+    def baz(x, y):
+        return x
+
+[file bar.py]
+from typing import Any
+any: Any
+
+from foo import F
+def bar() -> None:
+    F.bar(0, any)
+    F.baz(0, any)
```

Comment:
Maybe call with different types here, to make sure that separate signatures can be inferred? What about calling a static/class method through an instance instead of a type object?